### ISSUE TITLE
Error when vagrant mounting SMB shared folders

### ISSUE THREAD

**Issue by nasirouwagana** [puphpet/puphpet#1845]
Hello.

I am working on Windows 8.1 OS and today I am trying to create a virtual machine for my development with these tools : Vagrant 1.7.2 , VirtualBox 4.3.22, Puphpet.

I also used in my PuPHPet configuration SMB as shared folder type to improve the slowness of default windows shared folder.

Then, I encountered this issue when Vagrant trying to mount SMB Shared folders : 

```
D:\App\Boxes\project.dev                                                                                                             
λ vagrant reload                                                                                                                    
==> default: Attempting graceful shutdown of VM...                                                                                   
==> default: Checking if box 'puphpet/debian75-x64' is up to date...                                                                 
==> default: Clearing any previously set forwarded ports...                                                                          
==> default: Preparing SMB shared folders...                                                                                         
==> default: Clearing any previously set network interfaces...                                                                       
==> default: Preparing network interfaces based on configuration...                                                                  
    default: Adapter 1: nat                                                                                                          
    default: Adapter 2: hostonly                                                                                                     
==> default: Forwarding ports...                                                                                                     
    default: 22 => 2222 (adapter 1)                                                                                                  
==> default: Running 'pre-boot' VM customizations...                                                                                 
==> default: Booting VM...                                                                                                           
==> default: Waiting for machine to boot. This may take a few minutes...                                                             
    default: SSH address: 127.0.0.1:2222                                                                                             
    default: SSH username: vagrant                                                                                                   
    default: SSH auth method: private key                                                                                            
    default: Warning: Connection timeout. Retrying...                                                                                
==> default: Machine booted and ready!                                                                                               
GuestAdditions 4.3.22 running --- OK.                                                                                                
==> default: Checking for guest additions in VM...                                                                                   
==> default: Setting hostname...                                                                                                     
==> default: Configuring and enabling network interfaces...                                                                          
==> default: Mounting SMB shared folders...                                                                                          
    default: D:/Projects => /var/www                                                                                                 
Failed to mount folders in Linux guest. This is usually because                                                                      
the "vboxsf" file system is not available. Please verify that                                                                        
the guest additions are properly installed in the guest and                                                                          
can work properly. The command attempted was:                                                                                        

mount -t cifs -o uid=`id -u www-data`,gid=`getent group www-data | cut -d: -f3`,sec=ntlm,username=,pass=PASSWORDHIDDEN ///c373e080   
e30e1f7db166dd359eba98c2 /var/www                                                                                                    
mount -t cifs -o uid=`id -u www-data`,gid=`id -g www-data`,sec=ntlm,username=,pass=PASSWORDHIDDEN ///c373e080e30e1f7db166dd359eba9   
8c2 /var/www                                                                                                                         

The error output from the last command was:                                                                                          

stdin: is not a tty                                                                                                                  
mount: wrong fs type, bad option, bad superblock on ///c373e080e30e1f7db166dd359eba98c2,                                             
       missing codepage or helper program, or other error                                                                            
       (for several filesystems (e.g. nfs, cifs) you might                                                                           
       need a /sbin/mount.<type> helper program)                                                                                     
       In some cases useful info is found in syslog - try                                                                            
       dmesg | tail  or so 
```

  To solve my problem, I try these comands : 
1. vagrant plugin install vagrant-vbguest on my Host System (Windows 8.1 64bits)
2. sudo ln -s /opt/VBoxGuestAdditions-4.3.22/lib/VBoxGuestAdditions /usr/lib/VBoxGuestAdditions on my Guest System (Debian Wheezy 7 x64 )
3. vagrant relaod

Now I have always the same problem : Failed to mount folders in Linux guest.........

Please Help me to solve my problem.

Thank you.

**Comment by wimpyprogrammer** [comment]
I'm trying SMB for the first time and in a similar spot.  I think I've made some headway.

Here's my setup:
 - Windows Server 2012 R2 x64 host
 - VirtualBox 4.3.28 provider
 - Vagrant 1.6.3
 - Ubuntu Trusty 14.04 LTS x64 guest
 - [config.yaml](https://gist.github.com/wimpyprogrammer/6bca202bc994274102af) of a barebones box I just generated on puphpet.com

I also found the usual discussions of this problem which are focused on VirtualBox Guest Additions, per the "Failed to mount folders..." error message.  I installed `vagrant-vbguest` and the output of `vagrant up` showed that guest editions 4.3.28 is used.

Here's the output of `vagrant up` after adding `--debug` to `vagrantfile.vm.provision.puppet.options` in the `config.yaml`.  There's nothing too surprising.

```
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'puphpet/ubuntu1404-x64'...

[KProgress: 10%
[KProgress: 20%
[KProgress: 30%
[KProgress: 40%
[KProgress: 60%
[KProgress: 70%
[KProgress: 80%
[KProgress: 90%
[K==> default: Matching MAC address for NAT networking...
==> default: Checking if box 'puphpet/ubuntu1404-x64' is up to date...
==> default: Setting the name of the VM: smb-testing_default_1440821354770_68935
==> default: Preparing SMB shared folders...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
    default: Adapter 2: hostonly
==> default: Forwarding ports...
    default: 22 => 6259 (adapter 1)
    default: 22 => 2222 (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection timeout. Retrying...
==> default: Machine booted and ready!
GuestAdditions versions on your host (4.3.28) and guest (4.3.20) do not match.
stdin: is not a tty
Reading package lists...
Building dependency tree...
Reading state information...
linux-headers-3.13.0-32-generic is already the newest version.
Suggested packages:
  debhelper
The following NEW packages will be installed:
  dkms
0 upgraded, 1 newly installed, 0 to remove and 85 not upgraded.
Need to get 64.6 kB of archives.
After this operation, 349 kB of additional disk space will be used.
Err http://mirror.rackspace.com/ubuntu/ trusty-updates/main dkms all 2.2.0.3-1.1ubuntu5.14.04
  404  Not Found [IP: 72.4.120.219 80]
E: Failed to fetch http://mirror.rackspace.com/ubuntu/pool/main/d/dkms/dkms_2.2.0.3-1.1ubuntu5.14.04_all.deb  404  Not Found [IP: 72.4.120.219 80]

E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
stdin: is not a tty
Ign http://mirror.rackspace.com trusty InRelease
Ign http://security.ubuntu.com trusty-security InRelease
Ign http://mirror.rackspace.com trusty-updates InRelease
Get:1 http://security.ubuntu.com trusty-security Release.gpg [933 B]
Get:2 http://security.ubuntu.com trusty-security Release [63.5 kB]
Ign http://mirror.rackspace.com trusty-backports InRelease
Hit http://mirror.rackspace.com trusty Release.gpg
Get:3 http://mirror.rackspace.com trusty-updates Release.gpg [933 B]
Get:4 http://security.ubuntu.com trusty-security/main Sources [92.6 kB]
Get:5 http://mirror.rackspace.com trusty-backports Release.gpg [933 B]
Get:6 http://security.ubuntu.com trusty-security/restricted Sources [2,061 B]
Hit http://mirror.rackspace.com trusty Release
Get:7 http://security.ubuntu.com trusty-security/universe Sources [29.9 kB]
Get:8 http://mirror.rackspace.com trusty-updates Release [63.5 kB]
Get:9 http://security.ubuntu.com trusty-security/multiverse Sources [2,330 B]
Get:10 http://security.ubuntu.com trusty-security/main amd64 Packages [335 kB]
Get:11 http://mirror.rackspace.com trusty-backports Release [63.5 kB]
Get:12 http://security.ubuntu.com trusty-security/restricted amd64 Packages [8,875 B]
Get:13 http://security.ubuntu.com trusty-security/universe amd64 Packages [114 kB]
Hit http://mirror.rackspace.com trusty/main Sources
Hit http://mirror.rackspace.com trusty/restricted Sources
Get:14 http://security.ubuntu.com trusty-security/multiverse amd64 Packages [3,686 B]
Hit http://mirror.rackspace.com trusty/universe Sources
Get:15 http://security.ubuntu.com trusty-security/main i386 Packages [320 kB]
Hit http://mirror.rackspace.com trusty/multiverse Sources
Hit http://mirror.rackspace.com trusty/main amd64 Packages
Hit http://mirror.rackspace.com trusty/restricted amd64 Packages
Hit http://mirror.rackspace.com trusty/universe amd64 Packages
Get:16 http://security.ubuntu.com trusty-security/restricted i386 Packages [8,846 B]
Get:17 http://security.ubuntu.com trusty-security/universe i386 Packages [114 kB]
Hit http://mirror.rackspace.com trusty/multiverse amd64 Packages
Hit http://mirror.rackspace.com trusty/main i386 Packages
Get:18 http://security.ubuntu.com trusty-security/multiverse i386 Packages [3,830 B]
Get:19 http://security.ubuntu.com trusty-security/main Translation-en [182 kB]
Hit http://mirror.rackspace.com trusty/restricted i386 Packages
Hit http://mirror.rackspace.com trusty/universe i386 Packages
Get:20 http://security.ubuntu.com trusty-security/multiverse Translation-en [1,679 B]
Hit http://mirror.rackspace.com trusty/multiverse i386 Packages
Hit http://security.ubuntu.com trusty-security/restricted Translation-en
Hit http://mirror.rackspace.com trusty/main Translation-en
Get:21 http://security.ubuntu.com trusty-security/universe Translation-en [66.9 kB]
Hit http://mirror.rackspace.com trusty/multiverse Translation-en
Hit http://mirror.rackspace.com trusty/restricted Translation-en
Hit http://mirror.rackspace.com trusty/universe Translation-en
Get:22 http://mirror.rackspace.com trusty-updates/main Sources [231 kB]
Get:23 http://mirror.rackspace.com trusty-updates/restricted Sources [4,725 B]
Get:24 http://mirror.rackspace.com trusty-updates/universe Sources [134 kB]
Get:25 http://mirror.rackspace.com trusty-updates/multiverse Sources [5,143 B]
Get:26 http://mirror.rackspace.com trusty-updates/main amd64 Packages [609 kB]
Get:27 http://mirror.rackspace.com trusty-updates/restricted amd64 Packages [15.4 kB]
Get:28 http://mirror.rackspace.com trusty-updates/universe amd64 Packages [309 kB]
Get:29 http://mirror.rackspace.com trusty-updates/multiverse amd64 Packages [11.9 kB]
Get:30 http://mirror.rackspace.com trusty-updates/main i386 Packages [589 kB]
Get:31 http://mirror.rackspace.com trusty-updates/restricted i386 Packages [15.1 kB]
Get:32 http://mirror.rackspace.com trusty-updates/universe i386 Packages [310 kB]
Get:33 http://mirror.rackspace.com trusty-updates/multiverse i386 Packages [12.1 kB]
Get:34 http://mirror.rackspace.com trusty-updates/main Translation-en [293 kB]
Get:35 http://mirror.rackspace.com trusty-updates/multiverse Translation-en [6,148 B]
Get:36 http://mirror.rackspace.com trusty-updates/restricted Translation-en [3,560 B]
Get:37 http://mirror.rackspace.com trusty-updates/universe Translation-en [164 kB]
Get:38 http://mirror.rackspace.com trusty-backports/main Sources [5,860 B]
Get:39 http://mirror.rackspace.com trusty-backports/restricted Sources [28 B]
Get:40 http://mirror.rackspace.com trusty-backports/universe Sources [28.4 kB]
Get:41 http://mirror.rackspace.com trusty-backports/multiverse Sources [1,898 B]
Get:42 http://mirror.rackspace.com trusty-backports/main amd64 Packages [6,288 B]
Get:43 http://mirror.rackspace.com trusty-backports/restricted amd64 Packages [28 B]
Get:44 http://mirror.rackspace.com trusty-backports/universe amd64 Packages [32.6 kB]
Get:45 http://mirror.rackspace.com trusty-backports/multiverse amd64 Packages [1,571 B]
Get:46 http://mirror.rackspace.com trusty-backports/main i386 Packages [6,293 B]
Get:47 http://mirror.rackspace.com trusty-backports/restricted i386 Packages [28 B]
Get:48 http://mirror.rackspace.com trusty-backports/universe i386 Packages [32.6 kB]
Get:49 http://mirror.rackspace.com trusty-backports/multiverse i386 Packages [1,552 B]
Get:50 http://mirror.rackspace.com trusty-backports/main Translation-en [3,645 B]
Get:51 http://mirror.rackspace.com trusty-backports/multiverse Translation-en [1,215 B]
Hit http://mirror.rackspace.com trusty-backports/restricted Translation-en
Get:52 http://mirror.rackspace.com trusty-backports/universe Translation-en [29.6 kB]
Ign http://mirror.rackspace.com trusty/main Translation-en_US
Ign http://mirror.rackspace.com trusty/multiverse Translation-en_US
Ign http://mirror.rackspace.com trusty/restricted Translation-en_US
Ign http://mirror.rackspace.com trusty/universe Translation-en_US
Fetched 4,343 kB in 11s (372 kB/s)
Reading package lists...
stdin: is not a tty
Reading package lists...
Building dependency tree...
Reading state information...
linux-headers-3.13.0-32-generic is already the newest version.
Suggested packages:
  debhelper
The following NEW packages will be installed:
  dkms
0 upgraded, 1 newly installed, 0 to remove and 184 not upgraded.
Need to get 64.6 kB of archives.
After this operation, 349 kB of additional disk space will be used.
Get:1 http://mirror.rackspace.com/ubuntu/ trusty-updates/main dkms all 2.2.0.3-1.1ubuntu5.14.04.1 [64.6 kB]
dpkg-preconfigure: unable to re-open stdin: No such file or directory
Fetched 64.6 kB in 0s (217 kB/s)
Selecting previously unselected package dkms.
(Reading database ... 63614 files and directories currently installed.)
Preparing to unpack .../dkms_2.2.0.3-1.1ubuntu5.14.04.1_all.deb ...
Unpacking dkms (2.2.0.3-1.1ubuntu5.14.04.1) ...
Processing triggers for man-db (2.6.7.1-1) ...
Setting up dkms (2.2.0.3-1.1ubuntu5.14.04.1) ...
Copy iso file C:\Program Files/Oracle/VirtualBox/VBoxGuestAdditions.iso into the box /tmp/VBoxGuestAdditions.iso
stdin: is not a tty
mount: block device /tmp/VBoxGuestAdditions.iso is write-protected, mounting read-only
Installing Virtualbox Guest Additions 4.3.28 - guest version is 4.3.20
stdin: is not a tty
Verifying archive integrity... All good.
Uncompressing VirtualBox 4.3.28 Guest Additions for Linux............
VirtualBox Guest Additions installer
Removing installed version 4.3.20 of VirtualBox Guest Additions...
Copying additional installer modules ...
Installing additional modules ...
Removing existing VirtualBox DKMS kernel modules ...done.
Removing existing VirtualBox non-DKMS kernel modules ...done.
Building the VirtualBox Guest Additions kernel modules ...done.
Doing non-kernel setup of the Guest Additions ...done.
You should restart your guest to make sure the new modules are actually used

Installing the Window System drivers
Could not find the X.Org or XFree86 Window System, skipping.
An error occurred during installation of VirtualBox Guest Additions 4.3.28. Some functionality may not work as intended.
In most cases it is OK that the "Window System drivers" installation failed.
stdin: is not a tty
==> default: Checking for guest additions in VM...
==> default: Setting hostname...
==> default: Configuring and enabling network interfaces...
==> default: Mounting SMB shared folders...
    default: D:/Users/Andrew/Dropbox/Programming/Applications/smb-testing => /var/www
Failed to mount folders in Linux guest. This is usually because
the "vboxsf" file system is not available. Please verify that
the guest additions are properly installed in the guest and
can work properly. The command attempted was:

mount -t cifs -o uid=`id -u www-data`,gid=`getent group www-data | cut -d: -f3`,sec=ntlm,username=,pass=PASSWORDHIDDEN ///7079ecb55df57e2457e7e1b7c3c1e813 /var/www
mount -t cifs -o uid=`id -u www-data`,gid=`id -g www-data`,sec=ntlm,username=,pass=PASSWORDHIDDEN ///7079ecb55df57e2457e7e1b7c3c1e813 /var/www
```

Next I tried to SSH into my Ubuntu guest (succeeded) and run the `mount` commands at the end of the output above.  I found the following issues:

 1. When I run `ls /sbin/mount.*`, `cifs` is not listed.  I only see `fuse`, `lowntfs-3g`, `ntfs`, `ntfs-3g`, and `vboxsf`.  So I ran `sudo apt-get install cifs-utils` to add CIFS support [per the Ubuntu wiki](https://wiki.ubuntu.com/MountWindowsSharesPermanently#CIFS_installation).

 1. Next, the `mount` command fails because the `username=` parameter is present but blank and the SMB hostname is missing (`///sharename` is intended to be `//host/sharename`).  The PuPHPet GUI says it's OK to leave these fields blank and they are stored in the `config.yaml` as `''`.  If you delete the quotes (e.g. `smb_username: `) and run `vagrant reload`, SMB works great.  I'm prompted for my Windows username and password, and the IP address of my VirtualBox host is auto-detected.

@wnasirou can you try these steps on your system?  You might need to adapt some things for Debian and unfortunately I can't help much there (I can barely manage Ubuntu :smile:).  If you can confirm the problem, we can hope for a patch in PuPHPet.

**Comment by wimpyprogrammer** [comment]
Just noticed that problem 1 above was previously reported in #1837.

**Comment by alvinpascoe** [comment]
Thanks @wimpyprogrammer, step 2 alone seemed sufficient to help me access the drive.

**Comment by killua99** [comment]
This comment might help. https://github.com/puphpet/puphpet/issues/1930#issuecomment-166296966

**Comment by jtreminio** [comment]
Will be fixed by #1837
